subsys/bluetooth/controller/Kconfig

```diff
@@ -8,14 +8,18 @@ menu "Nordic BLE controller"
 
 config BLECTRL_MASTER_COUNT
 	int "Number of concurrent master roles"
-	default 1
+	default BT_MAX_CONN if (BT_CENTRAL && (!BT_PERIPHERAL))
```

Comment:
So if BT_CENTRAL && BT_PERIPHERAL, BLECTRL_MASTER_COUNT=BLECTRL_SLAVE_COUNT=1 ?
This is a bit tricky, and I feel it might be a bit "too clever".

I suppose the better fix is to actually upstream BT_MAX_MASTER_CONN and BT_MAX_SLAVE_CONN and rather use the sum of those for the Zephyr controller(s)?

Reply:
Maybe it is tricky, but it is just used to provide default value - to be changed manually if needed.

Upstreaming is always a long way, especially Zephyr is frozen now. And I don't think arithmetic dependencies can be used in Kconfig. If they could, BT_MAX_CONN could be set to the sum of BLECTRL_MASTER_COUNT and BLECTRL_SLAVE_COUNT directly instead of introducing strange conditions.

Reply:
@ulfalizer Is it possible to have arithmetic operations on Kconfig?
It would be nice if we could redefine BT_MAX_CONN downstream to be hidden and make it the sum of master and slave count instead when LL_NRFXLIB is selected. Is this possible?
Or maybe you have a better suggestion?

Reply:
Not possible to do arithmetic in Kconfig, unfortunately.

Could maybe have `BT_MAX_CONN` depend on `!LL_NRFXLIB` instead and just ignore the value when `LL_NRFXLIB` is selected.

Could just make it non-configurable with https://docs.zephyrproject.org/latest/guides/kconfig/index.html#optional-prompts as well, but not sure if it'd help here.

Reply:
`BT_MAX_CONN` is in `zephyr/subsys/bluetooth/Kconfig` unfortunately. We don't really can or want to introduce external symbols in Zephyr's Kconfig :/ We can however redefine the `BT_MAX_CONN` symbol in https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/subsys/bluetooth/Kconfig#L13. 

But we cannot ignore the `BT_MAX_CONN` symbol either as it's heavily used in the Bluetooth host and elsewhere. The cleanest option would be to find a way to set/overload
`BT_MAX_CONN = SLAVE_COUNT + MASTER_COUNT`
I've tried for a while to figure out ways achieving this within the limits of the Kconfig language, but have yet to succeed.

Some other options:
1. Remove, e.g., the `BLECTRL_SLAVE_COUNT` config and let it be

```c
#if (CONFIG_BT_MAX_CONN >= CONFIG_BLECTRL_MASTER_COUNT)
  #define BLECTRL_SLAVE_COUNT CONFIG_BT_MAX_CONN - CONFIG_BLECTRL_MASTER_COUNT
#else
  #error CONFIG_BLECTRL_MASTER_COUNT must be less than or equal to CONFIG_BT_MAX_CONN
#endif
```
in https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/subsys/bluetooth/controller/hci_driver.c.

2. Make master/slave count as fractions of `BT_MAX_CONN`.  `BT_MAX_CONN=4, MASTER_COUNT=1/2, SLAVE_COUNT=1-MASTER_COUNT`.

Any other ideas to resolve this with minimal user interaction are very welcome.
---
Just FYI, the reason for having separate SLAVE_COUNT and MASTER_COUNT is difference in memory usage per connection type/role. The Zephyr LL does not differentiate memory requirements for slave/master role.

Reply:
@thomasstenersen Thank you for the summary.
I think your option 1. looks like the best solution. But I would change to removing the master count. Simply because 1 slave N master is the normal use case, and N master plus M slave is the complex use case.
So then for the normal use-case there is one configurations required i.e. MAX_CONN
For the complex use-case there are two configurations, i.e MAX_CONN and SLAVE_COUNT.